src/main/java/org/mockito/internal/util/reflection/FieldInitializer.java

```diff
@@ -284,9 +279,17 @@ private int countMockableParams(Constructor<?> constructor) {
 
         @Override
         public FieldInitializationReport instantiate() {
+            Field[] fields = testClass.getClass().getDeclaredFields();
+            HashMap<String, Type> hashmap = new HashMap<>();
+
+            for (Field field : fields) {
+                hashmap.put(field.getName(), field.getGenericType());
+            }
             final MemberAccessor accessor = Plugins.getMemberAccessor();
             Constructor<?> constructor = biggestConstructor(field.getType());
-            final Object[] args = argResolver.resolveTypeInstances(constructor.getParameterTypes());
+            final Object[] args = argResolver.resolveTypeInstancesGeneric(
+                hashmap, constructor.getGenericParameterTypes());
+           //final Object[] args = argResolver.resolveTypeInstances(constructor.getParameterTypes());
```

Comment:
You are removing the only usage of `resolveTypeInstances` afaik, so instead of introducing a new method, let's fix the existing method.